internal/testing/exchange/exchange.go

```diff
@@ -157,3 +158,23 @@ func SetupWs(tb testing.TB, e exchange.IBotExchange) {
 
 	setupWsOnce[e] = true
 }
+
+var updatePairsMutex sync.Mutex
+var updatePairsOnce = make(map[exchange.IBotExchange]bool)
+
+// UpdatePairsOnce ensures pairs are only updated once in parallel tests
+func UpdatePairsOnce(tb testing.TB, e exchange.IBotExchange) {
+	tb.Helper()
+
+	updatePairsMutex.Lock()
+	defer updatePairsMutex.Unlock()
+
+	if _, ok := updatePairsOnce[e]; ok {
```

Comment:
```suggestion
	if updatePairsOnce[e] {
```

Reply:
On the fence about this, since I prefer existence tests most of the time.
OTOH, it doesn't hurt and it's clearer and this isn't a hot path.

Fixed ed22da9c